yeat/cli.py

```diff
@@ -8,39 +8,54 @@
 # -------------------------------------------------------------------------------------------------
 
 import argparse
+from argparse import Action
+import json
 from pathlib import Path
 from pkg_resources import resource_filename
 from snakemake import snakemake
+import sys
 import yeat
+from yeat.config import AssemblerConfig
 
 
-ASSEMBLY_ALGORITHMS = ["spades", "megahit", "unicycler"]
+CONFIG_TEMPLATE = [
+    dict(
+        algorithm="spades",
+        extra_args="--meta",
+    ),
+    dict(
+        algorithm="megahit",
+        extra_args="--min-count 5 --min-contig-len 300",
+    ),
+]
 
 
-def check_assemblers(assemblers):
-    algorithms = []
-    for algorithm in assemblers:
-        if algorithm not in ASSEMBLY_ALGORITHMS:
-            message = (
-                f"Found unsupported assembly algorithm with `--assemblers` flag: [[{algorithm}]]!"
-            )
-            raise ValueError(message)
-        if algorithm in algorithms:
-            message = (
-                f"Found duplicate assembly algorithm with `--assemblers` flag: [[{algorithm}]]!"
-            )
-            raise ValueError(message)
-        algorithms.append(algorithm)
+class InitAction(Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        json.dump(CONFIG_TEMPLATE, sys.stdout, indent=4)
+        print()
+        raise SystemExit()
 
 
-def run(read1, read2, assemblers, outdir=".", cores=1, sample="sample", dryrun="dry"):
+def run(read1, read2, algorithms, outdir=".", cores=1, sample="sample", dryrun="dry"):
```

Comment:
Hey man, I feel this. We joke often that naming things is one of the hardest problems in computer science, but as funny as it is it's true!

Both `assemblers` and `algorithms` are reasonable variable names here. It's a collection of `AssemblerConfig` objects, each of which has an `algorithm` attribute. So `configs` or `assembly_configs` would also be appropriate. Consider the suggested code in the comment below.

I'm definitely well into hair splitting territory by now, but `fastq{1,2}` might also be more appropriate names for the first two arguments than `read{1,2}`.